src/stssession/STSSessionController.ts

```diff
@@ -86,6 +96,16 @@ export class STSSessionController {
     this.awsAccessKeyId = '';
     this.awsSecretAccessKey = '';
     this.awsSessionToken = '';
+
+    // default api headers
+    this.httpHeaders = {
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    };
+
+    // collect credentials from api every 60 minutes
+    setInterval(() => this.credentialsFromAPI(), 1000 * 60 * 60);
```

Comment:
Should probably refresh before the credentials in order to give some buffer for getting and refreshing the token.

Reply:
This is a good idea; I'll get an early refresh in the code.